### DISCUSSION TITLE
Is there any way to re-apply the groups to the devices?

### DISCUSSION THREAD

**Discussion by cokefd** [Koenkk/zigbee2mqtt#8047] [General]
Hello,

I have around 70 lights, and work beautiful creating groups and managing them in that way.
Each light might be part of multiple groups for different purposes.
I recently changed the old power cables of my house, and I don't have light switches anymore.
If for some reason I need to rejoin a couple of lights, I would need to rejoin all the lights in a large area.
The problem is that they don't seems to remember the groups that they used to be part of, meaning that I would need to manually remove them from the group and add it again.

Is there any way to re-send the group information to all the devices?

Thank you in advance.

**Comment by sjorge** [comment]
You sort of can if you plan ahead. I recently repaired all my devices.
I did a backup of the `group` section of configuraiton.yaml, then via the frontend I removed all the devices I was repairing.
Enabled joining and added them again, stopped z2m and put back the backup of the group section I made earlier and started z2m again, on startup it tries to update groups on the devices to what is in the config. (Not you need to remove the devices first otherwise the database entry remains and z2m will still think they have the groups configured)

It's takes a few steps but it works rather well.

**Reply by cokefd** [reply]
Hello,

Thanks for the guidance. Seems like this works:
- make a copy of  "devices.yaml", "groups.yaml" and database.db
- re-join whatever devices you need to reset. In my case, are large sets of lights, so, if I want to reset a couple of bulbs, I need to do maybe 24 of them.
- stop z2m
- open "database.db" and remove all the group entries. Any editor with regex you can replace this with nothing:  .*"type":"Group".*\n
- start z2m. As you mentioned before, will see that the devices are not part of a group in the internal DB, so, will start populating them.
- I got a load of errors but on every restart adds more devices to groups, so, set a node-red flow to restart the z2m add-on every 90 seconds, and leave it for about an hour

Hope this helps
